user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -121,24 +149,50 @@ static void get_new_data(Gimbal_t *gimbal_data){
  */
 static void update_setpoints(Gimbal_t *gimbal_set){
     
-    gimbal_set->yaw_motor.pos_set += (-1) * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND) / 10;
-    gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 10;
+    if(gimbal_set->rc_update->rc.s[0] == RC_SW_MID || gimbal_set->rc_update->rc.s[0] == RC_SW_UP){
+        fp32 theta = -1 * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND)
+                * MOTOR_ECD_TO_RAD / 80.0f;
+        fp32 rotation[2] = {arm_cos_f32(theta), arm_sin_f32(theta)};
+        multiply_complex_a_by_b(gimbal_set->yaw_setpoint, rotation);
+        make_unit_length(gimbal_set->yaw_setpoint);
+
+        gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 80.0f;
+    }
     
-    int16_constrain(gimbal_set->yaw_motor.pos_set, YAW_MIN, YAW_MAX);
-    int16_constrain(gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
+    gimbal_set->pitch_motor.pos_set = int16_constrain(gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
+}
 
-    //TODO: worry about the case where pos_set is unsigned and rc channel manages to push it    
-    // negative for a moment, causing the position to wrap from below 0 to a maxvalue. 
+static fp32 a_dot_b(fp32 a[2], fp32 b[2]){
+    return a[0] * b[0] + a[1] * b[1];
 }
 
+static fp32 length_of_a_cross_b(fp32 a[2], fp32 b[2]){
+    return a[0] * b[1] - a[1] * b[0];
+}
+
+static int16_t get_error_sign(fp32 actual_position[2], fp32 set_position[2]){
+    return length_of_a_cross_b(actual_position, set_position) >= 0 ? 1 : -1;
+}    
+
+
 /** 
  * @brief  Increments PID loop based on latest setpoints and latest positions
  * @param  None
  * @retval None
  */
 static void increment_PID(Gimbal_t *gimbal_pid){
-    gimbal_pid->pitch_motor.current_out = PID_Calc(&gimbal_pid->pitch_motor.pid_controller, gimbal_pid->pitch_motor.pos_read, gimbal_pid->pitch_motor.pos_set);
-    gimbal_pid->yaw_motor.current_out = PID_Calc(&gimbal_pid->yaw_motor.pid_controller, gimbal_pid->yaw_motor.pos_read, gimbal_pid->yaw_motor.pos_set);
+    // error magnitude 1 * ERROR_MULTIPLIER corresponds to gimbal being 90 degrees out of place --> jump action observed by pedram
+    // TODO: consider revising PID to eliminate the ERROR_MULTIPLIER constant
+    
+    fp32 error = get_error_sign(gimbal_pid->yaw_position, gimbal_pid->yaw_setpoint) 
+                    * (1 - a_dot_b(gimbal_pid->yaw_position,gimbal_pid->yaw_setpoint))
```

Comment:
gimbal_pid->yaw_position,gimbal_pid->yaw_setpoint
Put a space here ----------^

Reply:
I think this sort of thing *can* be done by a reviewer

Reply:
Impossible